Increase visibility on mismatch between dof ordering and node ordering (#875)
Ferrite-FEM/Ferrite.jl: docs/src/topics/fe_intro.md

and denote it with $\Omega_h$. In this example the corners of the triangles are called
*nodes*.

> Next we introduce the finite element approximation $u_\mathrm{h} \approx u$ as a sum of (nodal)
> *shape functions*, where we denote each of these function by $\phi_i$ and the corresponding

Reviewer: ```suggestion
Next we introduce the finite element approximation $u_\mathrm{h} \approx u$ as linear combination of
*shape functions*, $\phi_i$, weighted by the corresponding *degree of freedoms* (dof) $\hat{u}_i$.
These are sometimes also called *weights* or *nodal values*, but the numbering of these do not correspond
to the node numbers in the grid in Ferrite. While such numbering is common in basic finite element codes,
Ferrite supports different approximations of the finite element fields and the geometry, prohibiting
such basic numbering. For more details, see the [Ferrite numbering rules](@ref "Ordering-of-Dofs").
```

> Next we introduce the finite element approximation $u_\mathrm{h} \approx u$ as linear combination of
> *shape functions*, $\phi_i$, weighted by the corresponding *degree of freedoms* (dof) $\hat{u}_i$.
> These are sometimes also called *weights* or *nodal values*, but the numbering of these do not correspond
> to the node numbers in the grid in Ferrite. While such numbering is common in basic finite element codes,

Author: ```suggestion
Next we introduce the finite element approximation $u_\mathrm{h} \approx u$ as linear combination of
*shape functions* $\phi_i$. The corresponding weights are usually called *degree of freedoms* (dof) $\hat{u}_i$.
In different contexts $\hat{u}_i$ is also just called *weights* or *nodal values*. The numbering of the dofs does not correspond
to the node numbers in the grid in Ferrite. While such numbering is common in basic finite element codes,
```

Reviewer: One more iteration ...
```suggestion
Next, we introduce the finite element approximation $u_\mathrm{h} \approx u$ as linear combination of
*shape functions*, $\phi_i$. The corresponding weights are usually called *degree of freedoms* (dofs), $\hat{u}_i$.
Sometimes, the dofs are called *weights* or *nodal values*. In Ferrite, the numbering of the dofs does not correspond
to the node numbers in the grid. While such numbering is common in basic finite element codes,
```

> *degree of freedom* (dof) $\hat{u}_i$. These are sometimes also called *weights* or *nodal values*.
> In general, the indices for the dofs do **not** coincide with the indices of the nodes of a grid,
> as we can use different shape functions for either. Hence, in Ferrite grid nodes are numbered
> differently than degrees of freedom by default.

Reviewer: Included above
```suggestion
```

Note that *shape functions* are sometimes referred to as *basis functions* or *trial functions*,
and instead of $\phi_i$ they are sometimes denoted $N_i$. In this example we choose to approximate

> the test function in the same way. This approach is known as the *(Bubnov-)Galerkin finite element

Reviewer: ```suggestion
the test function in the same way. This approach is known as the *Ritz-Galerkin finite element
```
Not sure, just you noted this somewhere else. Seems to me that Bubnov is a less restrictive classification, but in this example its even Ritz?

Author: I have always learned that
1. We call methods Ritz-Galerkin methods, if we utilize variational principles for energy minimization formulations
2. We call methods Bubnov-Galerkin methods, if we utilize approximate solutions of the weak form where the ansatz and test spaces coincide
3. We call methods Petrov-Galerkin methods, if <---....---> where the ansatz and test spaces are different
